ts/defs.bzl

```diff
@@ -318,19 +325,43 @@ def ts_project(
     typings_out_dir = declaration_dir if declaration_dir else out_dir
     tsbuildinfo_path = ts_build_info_file if ts_build_info_file else name + ".tsbuildinfo"
 
-    tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js, no_emit)
-    tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js, no_emit)
+    # Target names for tsc, dts+js transpilers
+    tsc_target_name = name
+    declarations_target_name = None
+    transpile_target_name = None
+
+    # typing outputs
+    tsc_typings_outs = []
+    tsc_typing_maps_outs = []
+    if not declaration_transpiler or declaration_transpiler == "tsc":
+        tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js, no_emit)
+        tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js, no_emit)
+    else:
```

Comment:
If `declaration_transpiler` is set and `transpiler` is `tsc`, should we assert that `declaration` is `False` and fail if it is not. We don't want the tsc "transpiler" also outputting declarations and trying to overwrite on the ones from the `declaration_transpiler` if not in the sandbox.

Reply:
Hmm, I don't think so... I think `declaration && declaration_transpiler != None` means we should invoke the `declaration_transpiler`, but add `--declaration false` when invoking tsc.

Just like when using `transpiler` we add `--emitDeclarationOnly` when invoking tsc.

Reply:
I've updated this...

`declaration: True` means declarations will be outputted... maybe from tsc, maybe from a `declaration_transpiler`

The `--emitDeclarationOnly`, `--declaration false` and `--noEmit` might be added to the `tsc` invocation depending on the transpiler+noemit rule attributes.